### ISSUE TITLE
Frame Application Screen: Implement Details Page 

### ISSUE THREAD

**Issue by lamara11** [amosproj/amos2024ss01-xcelerator-demo-app#54]
## User story

1. As a facility manager
2. I need a screen to view the details of facilities
3. So that  I can have an overlook of all details of each facilities

## Acceptance criteria

- [ ] The screen needs to have a chart for the pump data and one for the environment data 
- [ ] The screen needs to show an Eco Score window
- [ ] The screen needs to show the work orders and a link for new work orders
- [ ] The screen needs to have a function for locking/unlocking Building doors 

## Definition of done (DoD)

- [ ] Added only after week 5
- [ ] The same for all features
- [ ] Here goes the project specific part

## DoD general criteria

- [ ] Feature has been fully implemented
- [ ] Feature has been merged into the mainline
- [ ] All acceptance criteria were met
- [ ] Product owner approved features
- [ ] All tests are passing
- [ ] Developers agreed to release

## Dependencies

- Item [#53 ](https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/issues/53)

**Comment by lamara11** [comment]
![Facility_detailed_view](https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/assets/81018335/e071df1c-b23f-4f57-8961-e8b267fdef79)

**Comment by IngoSternberg** [comment]
## Definition of facilities and assets

@lamara11 I have some questions regarding the definition of facilites. Maybe also discuss with @Shahraz98 because I asked him about assets in the [work order issue](https://github.com/orgs/amosproj/projects/43/views/2?pane=issue&itemId=62262011)

Before now we just used this as our facility data

```
interface IFacilityMock {
	id: string;
	icon: string;
	notification: string; // text next to the icon
	heading: string;
	subheading: string;
	variant: string; // success(green)/warning(yellow)/critical(orange)
}
```

I would suggest we use the success/ warning/ critical for the states of the pump Igor mentioned. \
no flow = critical, flow but sometimes it drops to 0 = warning, working normal = success.

Apart from the id all this data is necessary to make the ix-push-card look good
![Image](https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/assets/166606862/030a4f50-6b8d-4038-b059-76a3dcd0b7a8)

Shahraz mentioned that every facility has assets. Is an asset f.e. a water pump? So does every facility have multiple assets and we need to pick the asset first before we can see a chart of the pump data. In that case we still need to talk about how you pick which asset is used or if we want to have a detail page for every asset you can navigate to from the facility detail view or something.

Of course we have to add the assets as data for the facility then.

## Diagram of the pump data

questions about the diagram for the POs and also @sn0wcat \
 In the Q&A I found pump data as an array of these elements
```
  {
      "MotorCurrent": 49.0,
      "PressureOut": 2.98391,
      "StuffingBoxTemperature": 25.0,
      "PressureIn": 1.0,
      "_time": "2024-05-02T01:58:51.478Z",
      "Flow": 245.29
  }
```

So we have MotorCurrent, PressureOut, StuffingBoxTemperature, PressureIn and Flow to display on the y axis relative to the time on the x axis. Would it be ok if there is always exactly one of these visible and the user selects which one he wants to see?

Another thing is from when to when we are supposed to show the time. Is the user able to select which moments he want to see? Or can he always just see the last 2 hours or something?

Is it necessary to update the chart it in real time, so data comes in every few seconds and you can see it moving? The alternative would be that it only changes every 10 minutes or so and you will just see a static chart (but it will still be updated every 10 minutes)


### Environment Data and Eco score diagram

Which data should be displayed here?

**Comment by IngoSternberg** [comment]
In the IP meeting today all my questions were resolved and I think I overworried a little

**Comment by IngoSternberg** [comment]
I think you misclicked @shahraz1998 I estimated the real size at 5

And just to document the details page looks as described (I even added a small popup window for the lock function so you can't change it on accident). The thing is that the charts still need some work (getting real data from the backend and then displaying all that data as multiple lines in the same chart), that will be done in the oncoming issues, which is why I estimated it lower. These things and some other small changes discussed with the IP will be done in a seperate issue since all the acceptance criteria listed here are met.

![grafik](https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/assets/166606862/806c458b-6532-4cd5-a431-95d90fbf8db6)
